Project-Descriptions-and-Plans/CV32E4_TBD/Project_Concept_for_CV32E4_TBD_Nov-27_2023.md

```diff
@@ -0,0 +1,250 @@
+* RVK zbkx (Crossbar permutation instructions) : xpem4 and xperm8 additionnal instructions may be supported, if relevant for DSP applications.
+* RVK zkn, zks, zkr and zkt are specific to cryptographic computations and will not be supported
+
+**Optionnal design changes**
+
+Introduce XPULPv3 custom extentions (preliminary, feature list may evolve):
+* ALU instructions that have a strict RVB/RVP equivalent opcode will be deprecated, new XPULPv3 opcodes will be RVB/RVP ones instead
+* Deprecate single operand *abs* instruction in favor of dual operand *pdif* (RVP superset of abs useful for manhattan distance computation)
+* Deprecate single operand *cv.cplxconj* instruction in favor of dual operand *pas* (RVP parallel add-sub)
+* Deprecate *cv.cplxmul.r* and *cv.cplxmul.i* in favor of a single instruction *cv.cplxmul* that computes both parts of complex multiplication
+* Introduce hermitian form of *cv.cplxmul* (rs1 x conj(rs2)) ? (widely used in DSP computation for telecom applications)
+* *list to be completed / updated*
+
+Add CV-X-IF interface:
+* End user product may require additionnal appliance-specific custom instructions, these instructions should be implemented in a co-processor.
+
+Floating Point Unit:
+* keep current CV-FPU integration or move it to CV-X-IF?
+
+Add CLIC support:
+* Core-local interrupt controller fits better with embedded real-time requirements than PLIC or any other (non RISC-V compliant)
+  external interrupt controller that requires a 2-step interrupt handling code.
+
+Design Portability:
+* Replace global clock gate by equivalent global enable signal
+  * this will make the design more "FPGA-friendly", or compliant with ASIC design rules that forbid any logic addition on clock tree
+  * clock gate cell model and/or simulator behaviour may introduce delta cycles on clock tree, leading to RTL/GL simulation mismatch
+* EDA tools parser bug work-arounds: current design coding style is standard-compliant, but EDA tools may
+  have bugs that prevent proper reading. Design should be parsed with several EDA tools and offending RTL code should be fixed.
+
+### Potential future design enhancements
+* remove deprecated features
+* Add Zce extension for further code-size reduction (not required for Multicore cluster).
+* Add bus attributes support.
+* move bitmanip, SIMD and complex number subsets of XPULP, RVB and RVP to a CV-X-IF compliant co-processor IP
+  * pros : this will make these features available to other processors, and would simplify verification work.
+  * cons : a performance loss is expected on instructions that use register pairs.
+
+## Who would make use of OpenHW output
+Companies needing more performances, less energy consumption or smaller code size for real-time control and/or DSP applications.
+
+## Initial Estimate of Timeline
+At Project Concept:
+* Preliminary RTL release expected in 2024 Q1
```

Comment:
That means RTL will be finished in 3-4 months, is that correct? Let's not call that release as I would anticipate we need to more than complete RTL prior to release

Reply:
No ! I'm expected RTL design to start in 2024 Q1. This is the expected date for our first push.

```suggestion
* Preliminary RTL code drop expected in 2024 Q1
```